hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -142,13 +142,19 @@ public void accept(MessageSendingResult result) {
                 handleFailedSending(message, result);
                 if (!isTtlExceeded(message) && shouldRetrySending(result)) {
                     retrySingleThreadExecutor.schedule(() -> retrySending(result),
-                            subscription.getSubscriptionPolicy().getMessageBackoff(), TimeUnit.MILLISECONDS);
+                            extractRetryDelay(result), TimeUnit.MILLISECONDS);
```

Comment:
this is not ok to schedule tasks using only time from received header:
- we should validate if value is within reasonable boundries (0, maxMessageBackoff)
- we should discard message if retry-after is after message TTL because we will never send this message anyway